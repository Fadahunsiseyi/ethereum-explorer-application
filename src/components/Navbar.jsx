import React from 'react';
import { useNavigate, useLocation } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { searchTransaction } from '../redux/search/SearchReducer';
import { ChevronLeftIcon, MicrophoneIcon, CogIcon } from '../assets/icons/Icons';

const Navbar = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const { pathname } = useLocation();
  const handleSearch = (e) => {
    dispatch(searchTransaction(e.target.value.trim()));
  };
  const handleNavigate = () => {
    navigate(-1);
  };

  return (
    <nav>
      <button type="button" className="chevBtn" onClick={handleNavigate}>
        <ChevronLeftIcon />
      </button>
      {pathname === '/' ? (
        <div className="search-input">
          <input
            name="search input"
            placeholder="Search Transaction..."
            id="search-input"
            onClick={handleSearch}
          />
        </div>
      ) : (
        <h3>
          Transaction Block Details
        </h3>
      ) }
      <div className="icons">
        <CogIcon />
        <MicrophoneIcon />
      </div>
    </nav>
  );
};

export default Navbar;
