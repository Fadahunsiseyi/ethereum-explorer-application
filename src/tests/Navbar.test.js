import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Navbar from '../components/Navbar';

describe('Test Navbar', () => {
  describe('Test App.js', () => {
    it('Test if the page loads', () => {
      const tree = renderer.create(
        <BrowserRouter>
          <Provider store={store}>
            <Navbar />
          </Provider>
        </BrowserRouter>,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
