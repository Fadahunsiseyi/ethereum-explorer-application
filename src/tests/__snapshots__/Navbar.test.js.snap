// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navbar Test App.js Test if the page loads 1`] = `
<nav>
  <button
    className="chevBtn"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="chevIcon"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15 19l-7-7 7-7"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </button>
  <div
    className="search-input"
  >
    <input
      id="search-input"
      name="search input"
      onClick={[Function]}
      placeholder="Search Transaction..."
    />
  </div>
  <div
    className="icons"
  >
    <svg
      className="cogIcon"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M11.49 3.17c-.38-1.56-2.6-1.56-2.98 0a1.532 1.532 0 01-2.286.948c-1.372-.836-2.942.734-2.106 2.106.54.886.061 2.042-.947 2.287-1.561.379-1.561 2.6 0 2.978a1.532 1.532 0 01.947 2.287c-.836 1.372.734 2.942 2.106 2.106a1.532 1.532 0 012.287.947c.379 1.561 2.6 1.561 2.978 0a1.533 1.533 0 012.287-.947c1.372.836 2.942-.734 2.106-2.106a1.533 1.533 0 01.947-2.287c1.561-.379 1.561-2.6 0-2.978a1.532 1.532 0 01-.947-2.287c.836-1.372-.734-2.942-2.106-2.106a1.532 1.532 0 01-2.287-.947zM10 13a3 3 0 100-6 3 3 0 000 6z"
        fillRule="evenodd"
      />
    </svg>
    <svg
      className="micIcon"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M7 4a3 3 0 016 0v4a3 3 0 11-6 0V4zm4 10.93A7.001 7.001 0 0017 8a1 1 0 10-2 0A5 5 0 015 8a1 1 0 00-2 0 7.001 7.001 0 006 6.93V17H6a1 1 0 100 2h8a1 1 0 100-2h-3v-2.07z"
        fillRule="evenodd"
      />
    </svg>
  </div>
</nav>
`;
